Contributing guide (#25)
Paperback-iOS/website: contributing.md

# Contributing

> You want to contribute to the website? Welcome!
> Our website runs off of VuePress. Here you will find contribution resources and advice.

If you have any questions, you can ask them in #website-dev channel in our Discord.

 * [Preparing the development environment](#preparing-the-development-environment)
 * [Guidelines](#guidelines-1)
 * [Localization](#localization)

## Preparing the development environment
### Fork the project
* Create your own fork of the repository by choosing the **Fork** option at the top right of the website github repository.
* Create a new branch before working on your new fork. The method of branching depends on the Git CLI/GUI that you use, reference your software manual, or contact us if you need help with this step.
* Once on your own branch, you're ready to start making improvements to the website

> > You can use a git client like [GitHub Desktop](https://desktop.github.com/) to manage your local repository.

Reviewer: Maybe more personal but I'm not a big fan of GD. I would suggest also recommending the two most popular visual Git clients as I think they're more beginner-friendly and have a nicer interface overall. GitKraken is the only one of these 3 that works natively on Linux too!

```suggestion
> You can use a git client like [GitHub Desktop](https://desktop.github.com/), [GitKraken](https://www.gitkraken.com/) or [Sourcetree](https://www.sourcetreeapp.com/) to manage your local repository.
```

Author: I agree with you and use [Git fork](https://git-fork.com/). The question is are these tools really more beginner friendly? 
I believe GD could not be simpler, it only has a fetch, push and commit buttons. On the other hand Git Kraken seems powerful and great but complex.

Linux support is also a good argument.
If you think it would be great to list a few choice for the reader and suggest other tools, I will be happy to add your suggestion.

When your done submit a pull request. This can be done by navigating to the main page of the [website git repository](https://github.com/Paperback-iOS/website) and selecting 'pull request' at the top.  You may have to click the button titled **compare across forks**. Ensure that the base repository belongs to Paperback-iOS, whereas the head is your fork, and your branch.

### Running the website
> When you make modifications to the site, you should validate your changes and formatting yourself first.

See [Building local enviroment](https://github.com/Paperback-iOS/website#building-local-enviroment)

## Guidelines
### Glossary

>  * App/application --> app
>  * Device/iPhone/iPad/phone --> iPhone or iPad
>  * iOS --> iOS or iPadOS

Reviewer: I think this will look nicer in a table

```suggestion
| Variations               | To be used    |
|:------------------------:|:-------------:|
| App/application          | app           |
| Device/iPhone/iPad/phone | iOS or iPadOS |
| iOS                      | iOS or iPadOS |
```

# Contributing

> You want to contribute to the website? Welcome!
> You will find here resources and advices that could help you.

If you have any questions, you can ask them in #website-dev channel in our Discord.

 * [Preparing the development environment](#preparing-the-development-environment)
 * [Guidelines](#guidelines-1)
 * [Localization](#localization)

<!-- /code_chunk_output -->

## Preparing the development environment
### Fork the project
* Fork the repo.
* **Always create a new branch before working on your fork.**
* You can now improve the website.

> You can use a git client like [GitHub Desktop](https://desktop.github.com/) to manage your local repository.

> When your done submit a pull request.

Reviewer 2: Be clear for people who haven't used Git before
```suggestion
When your done submit a pull request. This can be done by navigating to the main page of the [website git repository](https://github.com/Paperback-iOS/website) and selecting 'pull request' at the top.  You may have to click the button titled **compare across forks**. Ensure that the base repository belongs to Paperback-iOS, whereas the head is your fork, and your branch. ``

Author: Nice!
Done in ea456ca.

### Running the website

> > When you make modifications to the site it can be useful to render the code. You will thus see how the modifications look like.

Reviewer 2: ```suggestion
> When you make modifications to the site, you should validate your changes and formatting yourself first.
```

See [Building local enviroment](https://github.com/Paperback-iOS/website#building-local-enviroment)

## Guidelines
### Glossary

 * App/application --> app
 * Device/iPhone/iPad/phone --> iPhone or iPad
 * iOS --> iOS or iPadOS

> Be careful of capitalization:

* Paperback
* MangaDex

* AltStore
* AltServer

* Safari
* macOS

### Syntax and layout
> To create a numbered list use `1.` on all lines:
```markdown
1. Line 1
1. Line 2
```

> > Write `.IPA` in a code block.

Reviewer 2: Clarify a little more about this! Maybe give an example of what you mean exactly? Instructions are a little vague.

Author: I removed this line. People should not need to use `.IPA` often and if they use .IPA, we will tell them to change it to `.IPA`

> > Use `[Image]()` or `[GIF]()` to link to an explicative picture.

Reviewer 2: Give an example of such

```suggestion
> You may render images and gifs by enclosing it in the following format: `[http://yourimagelink.jpg]()` or `[http://yourgiflink.gif]()`
```

Author: That wasn't really what I meant. The idea was: you can link to a picture by using a  `[Image](http://yourimagelink.jpg)` or `[GIF](http://yourgiflink.gif)` link.
The objective was to use a consistent form of link for pictures: always [Image](http://yourimagelink.jpg) or [GIF](http://yourgiflink.gif).

I think we could remove this information with #30. The idea is to add a Vue component that create a dialog windows.

Reviewer 2: I like your additional component described in #32. I think I understand your intentions better now.
It's your call. I'd redact my suggestion, and suggest just sticking with what you've got initially if you don't decide to remove it.

Author: The component was added so this section is not needed anymore.

### "Not up to date" warning
> You've made a major modification to the English version of the site and you want to show that a translation is not up to date?
> You can  mark a page as "not up to date" by adding the following code at the top of the concerned page.

```html
<notuptodate/>
```
It will create a translated warning saying:

> > Major changes have been made in the English version of this page. We suggest you to use this one.

Reviewer 2: I understand that the warning isn't defined in this page, although the wording seems strange.

```suggestion
> Major changes have been made in the English version of this page. We suggest you use this revision instead.
```

Author: I used your more detailed suggestion in d708621:
```
Major changes have been made in the English version of this page. We suggest you use the English revision until this page is up to date.
```

### Components
You can use the following components. Try not to abuse of `warning`.
```markdown
::: tip Title
 Content
:::
```
```markdown
::: warning Title
 Content
:::
```
```markdown
:::: el-tabs
::: el-tab-pane label="Tab1"
Content1
:::
::: el-tab-pane label="Tab2"
Content2
:::
::::
```
```html
<el-tag type="warning">Not the Windows Store app!</el-tag>
```
```html
<Download text="from here"/>
```
```html
<notuptodate/>
```

> ### Potential issue

Reviewer 2: ```suggestion
### Potential issues
```

**Note that `tabs` currently needs to begin by a `h3` and finish by an `ol` components.** 
Otherwise there will be padding issues. See [PR #20](https://github.com/Paperback-iOS/website/pull/20)

## Localization
You want to translate the website?
The following languages are currently available:
|  Language  |  Translator  |    Reviewer    |
|:----------:|:------------:|:--------------:|
|   English  |              |                |
|   French   |     Lemon    |  Sirus, fma16  |
|  Italiano  |   Kinpatsu   |                |
| Nederlands |     Joery    |       Syn      |

In development: 
|  Language  |  Translator  |    Reviewer    |
|:----------:|:------------:|:--------------:|
|  Russian   |KorewaWatchful|     Dkzver     |

### Adding a new language
* Copy `website/src/README.md` to `website/src/LANG/README.md` and `website/src/help` to `website/src/LANG/help`.

> * Don't forget to edit `.vuepress` folder.

Reviewer 2: What should they be editing inside of the folder?

Author: I have try to improve this part with cfdb9b2. What do you think?

```markdown
### Adding a new language
 * Copy the following files:
   * `website/src/README.md` to `website/src/LANG/README.md`
   * `website/src/help` to `website/src/LANG/help`
   * `src/.vuepress/config/nav/en.js` to `src/.vuepress/config/nav/LANG.js`
   * `src/.vuepress/config/sidebar/en.js` to `src/.vuepress/config/sidebar/LANG.js`
 * Translate these files.
 * Edit `src/.vuepress/config.js` file the same way it was done for the other translations.

Don't forget to change links:
`/help/faq/#troubleshooting` -> `/fr/help/faq/#resolution-des-problemes-de-l-application`
```

Reviewer 2: I suggest you still be a little more clear about what you should be editing inside of `src/.vuepress/config.js` - You state 'the same way it was done for other translations' which is a little vague. The goal of documentation is to be explicit.

The rest of the instructions seem to be very clear!

Author: You're right!
I tried to explain more precisely what should be done in aa7f819.
I found it difficult to make it clear, so if you have some suggestions or feedback, I will be happy to add them.

 * [Guidelines](#guidelines-1)
 * [Localization](#localization)

## Preparing the development environment
### Fork the project
* Create your own fork of the repository by choosing the **Fork** option at the top right of the website github repository.
* Create a new branch before working on your new fork. The method of branching depends on the Git CLI/GUI that you use, reference your software manual, or contact us if you need help with this step.
* Once on your own branch, you're ready to start making improvements to the website

> You can use a git client like [GitHub Desktop](https://desktop.github.com/) to manage your local repository.

When your done submit a pull request. This can be done by navigating to the main page of the [website git repository](https://github.com/Paperback-iOS/website) and selecting 'pull request' at the top.  You may have to click the button titled **compare across forks**. Ensure that the base repository belongs to Paperback-iOS, whereas the head is your fork, and your branch.

### Running the website
> When you make modifications to the site, you should validate your changes and formatting yourself first.

> See [Building local enviroment](https://github.com/Paperback-iOS/website#building-local-enviroment)

Author: You're right, this would be better.
To be honest I'm a little lost with your suggestion... is this better than running `npm run serve`?

An other question, could `Node.js` be the right spelling?
```suggestion
**First time setup**
1. Open up a terminal window and navigate to your project root
1. Run the following command: `npm install`
  > You must have Node.js installed on your development environment, otherwise npm will not be a valid command
**Running the website**
1. Navigate to the project root in a terminal
1. Run the following command: `npx vuepress dev src`
  > npx is your node package executor. This allows you to run VuePress without installing it globally with npm or yarn.
```

> See [Building local enviroment](https://github.com/Paperback-iOS/website#building-local-enviroment)

Reviewer 2: Yeah, in hindsight your spelling is correct.
npm serve as a command may work just fine! If you’d rather swap it to that, that’s probably reasonable. I recall I had issue with that for some reason, so I listed out the full command.

My suggestion is simply what npm serve actually is doing.
